soccer/planning/Path.cpp

```diff
@@ -0,0 +1,36 @@
+#include "Path.hpp"
+#include <protobuf/LogFrame.pb.h>
+
+namespace Planning {
+
+// This method is a default implementation of draw() that works by evaluating
+// the path at fixed time intervals form t = 0 to t = duration.
+void Path::draw(SystemState* const state, const QColor& color,
+                const QString& layer) const {
+    Packet::DebugRobotPath* dbg = state->logFrame->add_debug_robot_paths();
+    dbg->set_layer(state->findDebugLayer(layer));
+
+    // Get the closest step size to a desired value that is divisible into the
+    // duration
+    const float duration = getDuration();
+    const float desiredStep =
+        0.25;  // draw the path by interpolating every x seconds
+    const float segmentCount = ceilf(duration / desiredStep);
+    const float step = duration / segmentCount;
```

Comment:
The way I have it, 0.25 is the initial *desired* step.  The step that's chosen is dependent on the result of `ceilf()`.  In your example, segmentCount would be 2, and `step` would get set to 0.125.  The loop would then run for i=0 and i=1, corresponding to t=0 and t=0.125.  Then the end would be added afterwards and there's no floating point ambiguity.